packages/language-server/src/server.ts

```diff
@@ -174,3 +225,60 @@ documents.onDidClose((e) => {
 documents.listen(connection);
 
 connection.listen();
+
+function isManifestDoc(uri: string): boolean {
+  return uri.endsWith("manifest.json");
+}
+
+function getFlagForXMLFile(uri: string): boolean {
+  const manifestFilesForCurrentFolder = Object.keys(
+    manifestData
+  ).filter((manifestUri) => uri.includes(dirname(manifestUri)));
+  const requiredManifestPath = maxBy(
+    manifestFilesForCurrentFolder,
+    (manifestFile) => manifestFile.length
+  );
+
+  if (requiredManifestPath === undefined) {
+    return false;
+  }
+
+  return manifestData[requiredManifestPath].isFlexEnabled;
+}
+
+async function getManifestDocumentsPromise(): Promise<string[]> {
```

Comment:
can `workspaceFolderPath` be passed here as an argument?
in that case we won't need it as moudle level state